astropy/nddata/tests/test_ccddata.py

```diff
@@ -1081,3 +1082,48 @@ def test_read_write_tilde_paths(home_is_tmpdir):
     # Ensure the unexpanded path doesn't exist (e.g. no directory whose name is
     # a literal ~ was created)
     assert not os.path.exists(filename)
+
+
+def test_ccddata_with_psf():
+    psf = _random_psf.copy()
+    ccd = CCDData(_random_array.copy(), unit=u.adu, psf=psf)
+    assert (ccd.psf == psf).all()
+
+    # cannot pass in non-ndarray
+    with pytest.raises(TypeError, match="The psf must be a numpy array."):
+        CCDData(_random_array.copy(), unit=u.adu, psf="something")
+
+
+def test_psf_setter():
+    psf = _random_psf.copy()
+    ccd = CCDData(_random_array.copy(), unit=u.adu)
+    ccd.psf = psf
+    assert (ccd.psf == psf).all()
+
+    # cannot set with non-ndarray
+    with pytest.raises(TypeError, match="The psf must be a numpy array."):
+        ccd.psf = "something"
```

Comment:
I think this would be a more realistic test case.
```suggestion
        ccd.psf = _random_array * u.pix
```